docs/proposals/scheduling/20250523-gpu-sharing-isolation-level.md

```diff
@@ -0,0 +1,79 @@
+---
+title: GPU Sharing Isolation Level
+authors:
+  - "@zqzten"
+reviewers:
+  - "@ZiMengSheng"
+  - "@hormes"
+creation-date: 2025-05-23
+last-updated: 2025-05-23
+---
+
+# GPU Sharing Isolation Level
+
+## Table of Contents
+
+<!-- TOC -->
+* [GPU Sharing Isolation Level](#gpu-sharing-isolation-level)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+  * [Proposal](#proposal)
+    * [User Story](#user-story)
+    * [API](#api)
+    * [Workflow](#workflow)
+    * [Feature Gate](#feature-gate)
+<!-- TOC -->
+
+## Summary
+
+Introduce a mechanism to differentiate a GPU share Pod's GPU resource isolation level. This takes effect on both scheduling and device allocating.
+
+## Motivation
+
+It's common that user want to use different GPU isolation levels for different GPU sharing workloads in one cluster. Currently, there're two common levels:
+
+- Isolated: Pod can only use (or see) the GPU resource it requested. e.g. The GPU has 16Gi memory and the Pod request 8Gi, it can only use up to 8Gi.
+- Shared: Pod can use (or see) the GPU resource up to the full card which assigned to it. e.g. The GPU has 16Gi memory and the Pod request 8Gi, it can use up to 16Gi.
+
+The Isolated is somewhat like K8s Guaranteed and the Shared is like Burstable.
+
+### Goals
+
+- Introduce a mechanism to differentiate a GPU share Pod's GPU resource isolation level.
+
+### Non-Goals
+
+- Introduce new GPU isolation providers.
+
+## Proposal
+
+### User Story
+
+User has two different kind of GPU sharing workloads in one K8s cluster. The first ones' GPU usage is predictable and user wants to ensure their stability, the second ones' GPU usage is burstable and user wants to maximize their GPU utilization.
+
+For the first ones, user labels the pods to set their GPU isolation level to **Isolated**, so that they would get guaranteed GPU share resources, different pods sharing the same GPU card would not disturb each other. Thus, the stability of the workload is ensured.
+
+For the second ones, user labels the pods to set their GPU isolation level to **Shared**, so that they would get burstable GPU share resources, every pod sharing the same GPU card can take use of the full card. Thus, the GPU utilization is maximized. A typical use case is running two pods with different models on the same GPU card, a model router would route requests to one of the model, so that there would be only one active inferencing process running on the GPU at the same time. By sharing the same GPU card, both of the pods could utilize the computing power of full card.
+
+### API
+
+- Introduce a new Pod label `koordinator.sh/gpu-share-isolation-level` whose value is one of `Isolated` and `Shared`. In the future, we might introduce more isolation levels such as `CoreIsolated` and `MemoryIsolated` for more use cases.
```

Comment:
This will complicate resource ops, as we need to look at the usage patterns of each GPU. Can this pattern be put at the node level to simplify management complexity?

Reply:
It can, but may reduce its flexibility. Sometimes user has few GPU machines and cannot afford to divide it at node level. Maybe we can provide best practice to users instead, that if conditions permit, it is recommended to divide different workloads with different usage patterns to different node pools.

Reply:
At least we need a place to see the current status of a GPU card.